### DISCUSSION TITLE
Is anyone else struggling with "positioning" in a combobox in React v9?

### DISCUSSION THREAD

**Discussion by vincentjflorio** [microsoft/fluentui#32047] [Q&A] [answered]
I appreciate that this is a known and default behavior for it to move itself around because otherwise the items would get clipped off-screen, but if the client wants to force it to stay put, the more material concern for me is none of the props, either string or shorthand, seem to affect the behavior at all:
![image](https://github.com/user-attachments/assets/dd3367b4-7d5f-400e-a62f-85c0a746f35b)

In the docs it makes mention of the existence of such a prop:
https://react.fluentui.dev/?path=/docs/components-combobox--default

But the old shorthands don't work:
https://react.fluentui.dev/?path=/docs/concepts-migration-from-v0-positioning--page
and none of these seem to apply to listboxes:
https://react.fluentui.dev/?path=/docs/concepts-developer-positioning-components--default

Would appreciate any clarification.

**Comment by khmakoto** [comment]
`Combobox` does have a `positioning` prop for you to pass the props that you can see in https://react.fluentui.dev/?path=/docs/concepts-developer-positioning-components--default. Have you tried using that?

**Reply by vincentjflorio** [reply]
Thanks for responding @khmakoto!

That links appears to be the same as the third link in my post. 

I may have misspoken when I said they don't seem to apply to listboxes. 

What I mean is, I tried adding them to a `Combobox`  *via* that `positioning` prop you mention, and regardless of the values supplied, it still sticks with the default behavior of moving back into the viewport as the user scrolls.

It's almost as if it cannot "hear" what we are saying.

**Reply by khmakoto** [reply]
@vincentjflorio what is it that you are trying to achieve? To force it to stay in one position even if you don't have enough space?

**Reply by vincentjflorio** [reply]
@khmakoto Yes; and believe me, I know it sounds weird, but it's what we were asked to do. I prefer it to be like you're doing, like react-menu, so someone can't scroll their choices (especially active ones) out of view where they can't interact with it anymore.

**Reply by khmakoto** [reply, marked as answer]
@vincentjflorio you need to use the `pinned` prop for this. You can see it working in the following [stackblitz](https://stackblitz.com/edit/59mdfe?file=package.json) by commenting and uncommenting the `pinned` prop.

This is what it looks like directly as a GIF.

__Not pinned:__
![NotPinned](https://github.com/user-attachments/assets/f1e846bc-5ea7-4c7e-9b54-b48702c33813)

__Pinned:__
![Pinned](https://github.com/user-attachments/assets/a84bee8a-2cb8-474d-9c14-d2c3f78f8aa4)

Hopefully that helps!

**Reply by vincentjflorio** [reply]
It does, and led me to some other relevant discoveries! Thank you for your amazingly fast response! I can't imagine how hands your full are right no, so it's very generous.

**Reply by khmakoto** [reply]
No problem! Glad I could help!

**Reply by mrmwink-wtwco** [reply]
@khmakoto -  I'm a bit late to the party on this but I'm having a similar issue and even with it pinned that causes a new problem that if you resize the screen as you show in the stackblitz, you can't actually scroll to see the items at the button of the list in order to select them, any ideas!?

**Reply by mrmwink-wtwco** [reply]
Ignore me, just found the following which fixes everything !!!

autoSize : true,

**Reply by khmakoto** [reply]
Just looking at this now @mrmwink-wtwco, but I'm happy that `autoSize` fixed your issue! Let me know if you need any more support!
